### DISCUSSION TITLE
Is there a way to delay setting the TCP port?

### DISCUSSION THREAD

**Discussion by Grauenwolf** [CoreWCF/CoreWCF#346] [Q&A]
In the sample, the TCP port for is set via `UseNetTcp` in `CreateWebHostBuilder`. 

The problem I'm running into is that I don't know which port I need until the `Startup` class is running and I have access to the `IConfiguration` object.

Is there a way to delay setting the delay setting the TCP port until `Startup.ConfigureServices` or `Startup.Configure` is running?

**Comment by Grauenwolf** [comment]
Looks like what I want is:

```
namespace CoreWCF.Configuration
{
    public static class ServiceModelWebHostBuilderExtensions
    {
        public static IServiceCollection AddNetTcp(this IServiceCollection services, int port)
        {
            services.TryAddEnumerable(ServiceDescriptor.Singleton<ITransportServiceBuilder, NetTcpTransportServiceBuilder>());
            services.AddSingleton(NetMessageFramingConnectionHandler.BuildAddressTable);
            services.AddNetTcpServices(new IPEndPoint(IPAddress.Any, port));
            services.AddTransient<IFramingConnectionHandshakeBuilder, FramingConnectionHandshakeBuilder>();

            return services;
        }
    }
}
```

This would have to be added to CoreWCF.NetTcp as it depends on internal classes.

**Reply by mconnew** [reply]
You can specify port 0, e.g. `.UseNetTcp(0)` and this will have Kestrel pick a port for you. You can fetch the port that was assigned using code similar to what is [here](https://github.com/CoreWCF/CoreWCF/blob/3374e762ab7ad4e02e71b15449a3901cf6567efa/src/CoreWCF.NetTcp/tests/Helpers/ServiceHelper.cs#L38);
